### PR TITLE
Upgrade chevrotain and resolve GraphQL parser ambiguities

### PR DESCRIPTION
**Pull request by jmrog**
As of chevrotain v4.7.0, the hierarchy of token categories is taken into consideration when a parser performs self-analysis. This reveals ambiguities in the parse rules for our GraphQL grammars. Since, in GraphQL, many tokens count as `Name` tokens, and since the `Name` category token is now taken into consideration by chevrotain during self-analysis, there are a number of duplicate prefix paths through the grammar.

This PR resolve the ambiguities (with a result being that our parsers can work with chevrotain v4.7.0+). The ambiguities are resolved by removing some special Stardog-specific additions to GraphQL from the _syntax_ level of the `StardogGraphQlParser`, and instead moving Stardog-specific additions (e.g., embedded SPARQL syntax) to the `StardogGraphQlVisitor` that is employed _after_ the parse phase. Currently, removing the additions from the syntax/parse level _does_ have a negative side effect, in that it eliminates some nice auto-completion for Stardog-specific GraphQL that we had working previously. To get that back, we'll have to move additional logic to the `StardogGraphQlVisitor`, but that will take some work.

### CODE REVIEW

**Review by joshhk72 — COMMENTED** [review]
Quick comment only

**Review by cchaffatt — COMMENTED** [review]
looked over briefly and will dive deeper in once out of draft.

chevrotain documentation now recommends we use perform self analysis as:
```
this.performSelfAnalysis()
```
and looks like they deprecated `Parser`.

Not necessary changes as of now but perhaps painless things we can add at the end.

**Review by joshhk72 — CHANGES_REQUESTED** [review]
I think things look good, all the prior features seem intact and new errors for directives seem to work pretty well. Few questions/comments for now

File: src/graphql/StardogGraphQlVisitor.ts [hunk 1]
```diff
@@ -2,156 +2,334 @@ import {
...
+          ...mapSparqlErrors(errors, argumentStringValueToken, 1)
+        );
+      }
+    }
+  });
+}
+
+function validateDirectiveArguments({
+  validators,
+  ...validatorOptions
```
Comment (joshhk72) [inline src/graphql/StardogGraphQlVisitor.ts:206]: Maybe it's better to just pass in `validatorOptions` as its own object

File: src/graphql/StardogGraphQlVisitor.ts [hunk 2]
```diff
@@ -2,156 +2,334 @@ import {
...
+  allowedArgumentTokenTypes,
+  directiveImage,
+  errorAccumulator,
+  numMinimumArguments,
+  suppliedArgumentNodes,
+}: ArgumentValidatorOptions) {
+  const numSuppliedArguments = suppliedArgumentNodes.length;
+  const numAllowedArguments = allowedArgumentTokenTypes.length;
+  const validArgumentsPhrase = `valid arguments: ${allowedArgumentTokenTypes
+    .map((argumentTokenType) => `\`${argumentTokenType.PATTERN}\``)
```
[the remark this answers is not in the document]
Comment (jmrog) [reply src/graphql/StardogGraphQlVisitor.ts:118]: This was _mostly_ just personal preference, but not entirely; the Monaco editor can also display markdown. We may want to use that feature on the Studio side to show the language keywords with special formatting. But I"m not wed to that idea, so could remove here if we want.

File: src/graphql/StardogGraphQlVisitor.ts [hunk 3]
```diff
@@ -2,156 +2,334 @@ import {
...
+      // again that we are currently *only* handling Stardog-specific
+      // directives that receive SPARQL here.
+      const argumentStringValueToken = getFirstChildCstElementByRuleStack(
+        ['Value', 'StringValue', 'StringValueToken'],
+        argumentNode
+      );
+
+      if (!isIToken(argumentStringValueToken)) {
+        // More safety; should not be a real possibility.
+        return;
```
[the remark this answers is not in the document]
Comment (jmrog) [reply src/graphql/StardogGraphQlVisitor.ts:184]: Yes, you're right; I think this comment is a leftover that should have been removed when I refactored. I'll update it.

Re: the tangential question: yes. :)

File: src/graphql/StardogGraphQlVisitor.ts [hunk 4]
```diff
@@ -2,156 +2,334 @@ import {
...
+      const [directiveNameToken] = ctx.Name as IToken[];
 
-      if (!stringValueNode.children.StringValueToken) {
-        // A bind directive can be identified at times by the parser even when
-        // there is no StringValueToken yet, due to error recovery.
+      // NOTE: We currently only provide special handling for Stardog-specific
+      // directives that accept SPARQL, so we bail early in all other cases.
+      // We may want to expand this later.
+      if (
+        !directiveNameToken ||
```
[the remark this answers is not in the document]
Comment (jmrog) [reply src/graphql/StardogGraphQlVisitor.ts:290]: Yep, this is also a leftover thing from refactoring. Removing.

File: src/graphql/StardogGraphQlVisitor.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshhk72) [reply src/graphql/StardogGraphQlVisitor.ts:118]: Seems good as it is.

**Review by joshhk72 — APPROVED** [review]
LGTM, but want to take a quick look again

**Comment by joshhk72** [comment]
LGTM (again) on second glance through

**Pull request merged**